airflow/providers/google/CHANGELOG.rst

```diff
@@ -27,10 +27,20 @@
 Changelog
 ---------
 
+10.21.0
+.......
+
+.. note::
+
+  The change  ``Fix 'GCSToGCSOperator' behavior difference for moving single object (#40162)`` has
+  been reverted as it turned out to break too much existing workflows. The behavior of the
+  ``GCSToGCSOperator`` has been restored to the previous behavior.
```

Comment:
Moving forward is the change that we are reverting is something that we actually want?
 we can revert and cut a bug fix release and right after redo the PR and cut a breaking change release

Reply:
I'd say at least not how it was prepared in that PR, as far as I understand now there is no way to rename files, copy single files, and other cases were not considered.

Reply:
Yeah. It is just not done "right" for all cases. So yeah - we could add bugfix release for that one, but I think since the last release we already have some features added like this https://github.com/apache/airflow/issues/40556 - so regardless if it contains bugfix, the next version will be `minor` bump.

Reply:
In that case @VladaZakharova @MaksYermak we should add a test to cover this part. If we know a specific behaviour is undesired then we should defiance against it. No one will remember that we had this issue if similar PR will be raised few months from now.

Reply:
@eladkal makes sense, I will prepare PR today with a unit test to cover this undesired behaviour.

Reply:
@MaksYermak please also improve the system test (`example_gcs_to_gcs.py`), currently we just run the operator but don't check if the destination file is what we expect, we could at least check if the file exist in the expected path.

Reply:
Cool :)